src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,112 @@
+package duke;
+
+import duke.task.Task;
+import duke.task.TaskList;
+
+/**
+ * The Ui class handles user interface-related operations and generates user-friendly messages for the Duke chatbot.
+ *
+ * @author selwyn
+ */
+public class Ui {
+    private static final String NAME = "DUKE PRINCE";
+
+    /**
+     * Generates a representation of the task list, optionally filtered by a search query.
+     *
+     * @param taskList   The task list to be represented.
+     * @param isMatching Whether the task list is filtered by a search query.
+     * @return A string representation of the task list.
+     */
+    public static String printTaskList(TaskList taskList, boolean isMatching) {
+        return taskList.taskListRepresentation(isMatching);
+    }
+
+    /**
+     * Generates a message indicating that a task has been successfully added to the task list.
+     *
+     * @param addedTask            The task that has been added.
+     * @param noOfTasksAfterAdding The number of tasks in the list after adding the new task.
+     * @return A message indicating the successful addition of the task.
+     */
+    public static String printAddedTask(Task addedTask, int noOfTasksAfterAdding) {
```

Comment:
Perhaps "noOfTasksAfterAdding" could be named "numOfTasksAfterAdding" for clarity and consistency.